sled-agent/src/rack_setup/service.rs

```diff
@@ -0,0 +1,227 @@
+            .expect("Cannot turn config to string");
+
+        // First, check if this request has previously been made.
+        //
+        // Normally, the rack setup service is run with a human-in-the-loop,
+        // but with this automated injection, we need a way to determine the
+        // (destructive) initialization has occurred.
+        //
+        // We do this by storing the configuration at "rss_config_path"
+        // after successfully performing initialization.
+        let rss_config_path = std::path::Path::new(crate::OMICRON_CONFIG_PATH)
+            .join("config-rss.toml");
+        if rss_config_path.exists() {
+            info!(
+                self.log,
+                "RSS configuration already exists at {}",
+                rss_config_path.to_string_lossy()
+            );
+            let old_config: Config = toml::from_str(
+                &tokio::fs::read_to_string(&rss_config_path).await?,
+            )?;
+            if &old_config == config {
+                info!(
+                    self.log,
+                    "RSS config already applied from: {}",
+                    rss_config_path.to_string_lossy()
+                );
+                return Ok(());
+            }
+
+            // TODO(https://github.com/oxidecomputer/omicron/issues/724):
+            // We could potentially handle this case by deleting all
+            // partitions (in preparation for applying the new
+            // configuration), but at the moment it's an error.
+            warn!(
+                self.log,
+                "Rack Setup Service Config was already applied, but has changed.
+                This means that you may have partitions set up on this sled, but they
+                may not match the ones requested by the supplied configuration.\n
+                To re-initialize this sled:
+                   - Disable all Oxide services
+                   - Delete all partitions within the attached zpool
+                   - Delete the configuration file ({})
+                   - Restart the sled agent",
+                rss_config_path.to_string_lossy()
+            );
+            return Err(SetupServiceError::Configuration);
+        } else {
+            info!(
+                self.log,
+                "No RSS configuration found at {}",
+                rss_config_path.to_string_lossy()
+            );
+        }
+
+        // Issue the dataset initialization requests to all sleds.
+        futures::future::join_all(
+            config.requests.iter().map(|request| async move {
+                info!(self.log, "observing request: {:#?}", request);
+                let dur = std::time::Duration::from_secs(60);
+                let client = reqwest::ClientBuilder::new()
+                    .connect_timeout(dur)
+                    .timeout(dur)
+                    .build()
+                    .map_err(|e| nexus_client::Error::<nexus_client::types::Error>::from(e))?;
```

Comment:
How is this related to Nexus? IIUC, this is a stand-in for the request to a sled, from the RSS. Where is Nexus in the loop here, either at this point, or in the future?

Reply:
Good question. I think this error conversion is wrong, but just flexible enough to compile (misleadingly.

I'll propagate the underlying error (`reqwest::Error`) instead.

Reply:
Sweet, that makes sense. In that case, I think the `NexusApi` error variant can also go away. I don't see it used elsewhere, but I could be wrong.